### ISSUE TITLE
kelnerbase.Kelner.getpass add stream parameter

### ISSUE THREAD

**Issue by MarekLew** [ipython/ipykernel#192]
`kelnerbase.Kelner.getpass` not have stream parameter.

I use jupyter notebook(4.0.1) and try use fabric library in `jupyter notebook` page. 
I have problem:

```
.../local/lib/python2.7/site-packages/Fabric-1.12.0-py2.7.egg/fabric/network.pyc in _password_prompt(prompt, stream)
    618     # NOTE: Using encode-to-ascii to prevent (Windows, at least) getpass from
    619     # choking if given Unicode.
--> 620     return getpass.getpass(prompt.encode('ascii', 'ignore'), stream)
    621 
    622 def prompt_for_password(prompt=None, no_colon=False, stream=None):

TypeError: getpass() takes at most 2 arguments (3 given)
```

stream is not useless in notebook, it`s only used to write prompt, and in this case must only passed.

Proposal:

```
    def getpass(self, prompt='', stream=None):
        """Forward getpass to frontends

        Param `stream` is ignored 

        Raises
        ------
        StdinNotImplentedError if active frontend doesn't support stdin.
        """
        if not self._allow_stdin:
            raise StdinNotImplementedError(
                "getpass was called, but this frontend does not support input requests."
            )
        return self._input_request(prompt,
            self._parent_ident,
            self._parent_header,
            password=True,
        )
```

**Comment by minrk** [comment]
We certainly should support getpass's actual signature.

**Comment by minrk** [comment]
closed by #201
